src/wasm/wasm-binary.cpp

```diff
@@ -2221,11 +2221,11 @@ void WasmBinaryBuilder::visitBlock(Block *curr) {
   }
 }
 
-Expression* WasmBinaryBuilder::getMaybeBlock(WasmType type) {
+Expression* WasmBinaryBuilder::getList(WasmType type) {
   auto start = expressionStack.size();
   processExpressions();
   size_t end = expressionStack.size();
-  if (start - end == 1) {
+  if (end - start == 1) {
```

Comment:
amusingly this was wrong all along, so the optimization never kicked in (if it did, it would have led to breakage, which is fixed in this PR - we need to call getBlock in places where a branch might occur, like the top of a function)